temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowThreadContext.java

```diff
@@ -78,25 +78,32 @@ public void yield(String reason, Supplier<Boolean> unblockFunction) {
     // Evaluates unblockFunction out of the lock to avoid deadlocks.
     lock.lock();
     try {
+      if (destroyRequested) {
+        throw new DestroyWorkflowThreadError();
+      }
+      yieldReason = reason;
+
       // TODO: Verify that calling unblockFunction under the lock is a sane thing to do.
       while (!inRunUntilBlocked || !unblockFunction.get()) {
-        if (destroyRequested) {
-          throw new DestroyWorkflowThreadError();
-        }
         status = Status.YIELDED;
         runCondition.signal();
         yieldCondition.await();
+        if (destroyRequested) {
+          throw new DestroyWorkflowThreadError();
+        }
         maybeEvaluateLocked(reason);
-        yieldReason = reason;
       }
```

Comment:
Is it impossible for `reason` to change in the loop b/c lock is held? Curious why `yieldReason = reason;` got lifted out of the loop

Reply:
How? It's a parameter of the method and we don't change it.

Reply:
Ah I didn't see it was even a param. Makes sense.